File: src/libs/datas/standings.js
import { WebData } from './data';

/**
 * コンテストの順位表データを取得し、他のタブと同期的にデータを扱います。
 */
export class StandingsData extends WebData {
    constructor(contestScreenName) {
        super(
            `https://atcoder.jp/contests/${contestScreenName}/standings/json`
        );
    }
}

File: crawler/ac_predictor_crawler/client/standings.py
from ac_predictor_crawler.logger import logger
from ac_predictor_crawler.requests import get_atcodersession

def get_standings(contest_screen_name: str):
  session = get_atcodersession()
  res = session.get(f"/contests/{contest_screen_name}/standings/json")
  session.check_logged_in()
  if not res.ok:
    logger.error(f"failed to get standings ({res.reason}), {contest_screen_name=}")
    exit(1)
  return res.json()
